### PR TITLE
feat: refactor app-tools

### PR DESCRIPTION
**Pull request by caohuilin**
## Summary

- add checkEntryPoint hook to move check entry point to runtime plugin
- add modifyEntrypoints hook to move entry point info to runtime plugin
- refactor app tools command define
- move router handler to router plugin

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/solutions/app-tools/src/plugins/analyze/getBundleEntry.ts [hunk 1]
```diff
@@ -60,53 +62,40 @@ export const getBundleEntry = (
...
+            : value.customBootstrap &&
+              ensureAbsolutePath(appDirectory, value.customBootstrap),
+        fileSystemRoutes: fs
+          .statSync(ensureAbsolutePath(appDirectory, entryName))
+          .isDirectory()
+          ? {}
+          : undefined,
+      };
 
       if (entrypoint.fileSystemRoutes && !isRouterV5(config)) {
```
Comment (zllkjc) [inline packages/solutions/app-tools/src/plugins/analyze/getBundleEntry.ts:95]: should we put this logic in router plugin ?

File: packages/runtime/plugin-runtime/package.json [hunk 2]
```diff
@@ -93,6 +93,11 @@
       "jsnext:source": "./src/router/runtime/server.ts",
       "default": "./dist/esm/router/runtime/server.js"
     },
+    "./router/cli": {
```
Comment (zllkjc) [inline packages/runtime/plugin-runtime/package.json:96]: Where did we use `@modern-js/runtime/router/cli`?

File: packages/runtime/plugin-runtime/src/cli/index.ts [hunk 3]
```diff
@@ -4,6 +4,9 @@ import { statePlugin } from '../state/cli';
 import { ssrPlugin } from '../ssr/cli';
 import { routerPlugin } from '../router/cli';
 import { documentPlugin } from '../document/cli';
+import { isRuntimeEntry } from './entry';
+
+export { isRuntimeEntry } from './entry';
```
Comment (zllkjc) [inline packages/runtime/plugin-runtime/src/cli/index.ts:9]: why exported here

File: packages/runtime/plugin-runtime/src/router/cli/handler.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+export async function handleModifyEntrypoints(
+  api: PluginAPI<AppTools<'shared'>>,
+  entrypoints: Entrypoint[],
+) {
+  const newEntryPoints = modifyEntrypoints(entrypoints);
+  const appContext = api.useAppContext();
+  const resolvedConfig = api.useResolvedConfigContext();
+  appContext.entrypoints = newEntryPoints;
+  originEntrypoints = cloneDeep(newEntryPoints);
+  const { generateCode } = await import('./code');
```
Comment (zllkjc) [inline packages/runtime/plugin-runtime/src/router/cli/handler.ts:20]: Perhaps in the future we should remove the generated code logic from `ModifyEntryPoints`..put in separate place

File: packages/runtime/plugin-runtime/src/router/cli/handler.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    pagesDir.some(pageDir => name.includes(pageDir));
+
+  const absoluteFilePath = path.resolve(appDirectory, filename);
+  const isRouteComponent =
+    isPageFile(absoluteFilePath) && isPageComponentFile(absoluteFilePath);
+
+  if (isRouteComponent && (eventType === 'add' || eventType === 'unlink')) {
+    const resolvedConfig = api.useResolvedConfigContext();
+    const { generateCode } = await import('./code');
+    const entrypoints = cloneDeep(originEntrypoints);
```
Comment (zllkjc) [inline packages/runtime/plugin-runtime/src/router/cli/handler.ts:50]: Why we need `cloneDeep` one more time, what happended if we remove this clone.. @yimingjfe

File: packages/runtime/plugin-runtime/package.json (hunk 2 above)
Comment (caohuilin) [reply packages/runtime/plugin-runtime/package.json:96]: in inside, router plugin need use router function. The export here can be removed after the plugin modifications are completed.

File: packages/solutions/app-tools/src/plugins/analyze/getFileSystemEntry.ts [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        return {
+          entryName,
+          isMainEntry: false,
+          entry: entryFile,
+          absoluteEntryDir: path.resolve(dir),
+          isAutoMount: true,
+          customBootstrap,
+        };
+      }
+      return {
```
[the remark this answers is not in the document]
Comment (caohuilin) [reply packages/solutions/app-tools/src/plugins/analyze/getFileSystemEntry.ts:70]: 嗯 确实3这个逻辑不存在

File: packages/runtime/plugin-runtime/src/router/cli/handler.ts (hunk 4 above)
Comment (caohuilin) [reply packages/runtime/plugin-runtime/src/router/cli/handler.ts:20]: yeap

File: packages/runtime/plugin-runtime/package.json (hunk 2 above)
Comment (zllkjc) [reply packages/runtime/plugin-runtime/package.json:96]: Ok, i see it~

**Review by zllkjc — APPROVED** [review]

**Pull request merged**